### PR TITLE
enh(Autocomplete): Add end adornment to Autocomplete

### PR DESCRIPTION
**Pull request by Thebarda**
singleAdornment
singleAdornmentMulti

### CODE REVIEW

**Comment by Thebarda** [comment]
Material-ui updates Autocomplete props type very recently: https://github.com/mui-org/material-ui/issues/21185

**Comment by bdauria** [comment]
@Thebarda can you detail the needs for this change?

**Comment by Thebarda** [comment]
> @Thebarda can you detail the needs for this change?

For example, inside credentials form part of job wizard (Auto Disco), we have an end adornment to edit selected credentials.

cred

File: src/InputField/Select/Autocomplete/index.tsx [hunk 1]
```diff
@@ -36,20 +43,29 @@ const LoadingIndicator = (): JSX.Element => {
...
+> &
+  UseAutocompleteProps<SelectEntry, Multiple, DisableClearable, FreeSolo>;
 
 const AutocompleteField = ({
   options,
   label,
   placeholder = '',
   loading = false,
   onTextChange = (): void => undefined,
+  endAdornmentInput = undefined,
```
Comment (bdauria) [inline src/InputField/Select/Autocomplete/index.tsx:68]:
```suggestion
  endAdornment = undefined,
```

File: src/InputField/Select/Autocomplete/index.tsx [hunk 2]
```diff
@@ -36,20 +43,29 @@ const LoadingIndicator = (): JSX.Element => {
...
+type FreeSolo = boolean;
+
 export type Props = {
   loading?: boolean;
   onTextChange?;
   label: string;
   placeholder?: string;
-} & Omit<AutocompleteProps<SelectEntry>, 'renderInput'> &
-  UseAutocompleteProps<SelectEntry>;
+  endAdornmentInput?: React.ReactElement;
```
Comment (bdauria) [inline src/InputField/Select/Autocomplete/index.tsx:55]:
```suggestion
  endAdornment?: React.ReactElement;
```

File: src/InputField/Select/Autocomplete/index.tsx [hunk 3]
```diff
@@ -70,6 +86,24 @@ const AutocompleteField = ({
...
+          }}
+          InputProps={{
+            ...params.InputProps,
+            endAdornment: (
+              <>
+                <InputAdornment
+                  classes={{ root: classes.inputEndAdornment }}
+                  position="end"
+                >
+                  {endAdornmentInput}
```
Comment (bdauria) [inline src/InputField/Select/Autocomplete/index.tsx:101]:
```suggestion
                  {endAdornment}
```

**Pull request merged**